integration-tests/tests/run-workflow.js

```diff
@@ -17,6 +17,8 @@ const testRunWorkflowFn = _.flow(
   await dismissNotifications(page)
 
   await createEntityInWorkspace(page, billingProject, workspaceName, testEntity)
+  // Wait for bucket access to avoid sporadic failure when launching workflow.
+  await checkBucketAccess(page, billingProject, workspaceName)
```

Comment:
Another option would be to call this from `makeWorkspace` (in `integration-helpers.js`). There are other tests that have shown similar failures. For example, running a notebook requires loading the notebooks tab which lists the current notebooks which fails if the bucket can't be read.

Reply:
I was wondering if we should make it in a more central location. My concern though is that we want to make the integration tests as close to a user experience as possible. In general, this problem is indicative of something that actually does impact users, and I would like us to consider on a test-by-test basis if the fix should be something in the product code itself (to handle the failure condition better), or if it is OK to make the test do something that is different from what the user would experience.

Reply:
Also, it may not be true that every time you create a workspace you need to wait for the write access, and I don't wish to change the timing in general.

Reply:
Thank you both for this excellent discussion.